src/plist.h

```diff
@@ -314,35 +314,54 @@ EXPORT_INLINE Obj NewImmutableEmptyPlist(void)
 /****************************************************************************
 **
 *F  NewPlistFromArray(<list>,<length>) . . create a plain list from a C array
+*F  NewPlistFromArgs(<args...>) .  create a plain list from list of arguments
+**
+**  This macro turns a variable-length list of macro arguments into an array,
+**  which is then passed to NewPlistFromArray.
+**
+**  __VA_ARGS__ contains the list of arguments given to this macro.
+**  There is no (nice) way to convert this list of arguments to an array
+**  in both C and C++, so we need two implementations
 */
-EXPORT_INLINE Obj NewPlistFromArray(const Obj * list, Int length)
+
+#ifdef __cplusplus
+} // escape extern "C"
+template <unsigned int length>
+EXPORT_INLINE Obj NewPlistFromArrayCpp(const Obj (&list)[length])
 {
     if (length == 0) {
```

Comment:
Note that strictly spealing, length zero arrays are not allowed, so the code two lines above might give an error. Also, `__VA_ARGS__` is not allowed to be empty according to the standard (although GNU C and some other compilers support it as an extension)

Reply:
That's very hard to reasonably fix (although it would be easy to fix in C++11, so let's consider that if it ever becomes a problem)